pkg/controller/mpi_job_controller.go

```diff
@@ -750,6 +754,18 @@ func (c *MPIJobController) getRunningWorkerPods(mpiJob *kubeflow.MPIJob) ([]*cor
 	return podList, nil
 }
 
+func (c *MPIJobController) countReadyWorkerPods(workers []*corev1.Pod) (int) {
+	ready := 0
+	for _, pod := range workers {
+		for _, c := range pod.Status.Conditions {
+			if c.Type == corev1.PodReady && c.Status == corev1.ConditionTrue {
+				ready++
```

Comment:
```suggestion
				ready++
				break
```

We don't need to check the rest of the conditions

Reply:
fixed this one